src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -803,6 +808,19 @@ public String getOneofVariantTypeName(OneofConfig oneof) {
     return getNotImplementedString("SurfaceNamer.getOneofVariantTypeName");
   }
 
+  /**
+   * The formatted name of a type used in long running operations, i.e. the operation payload and
+   * metadata,
+   */
+  public String getLongRunningOperationTypeName(ModelTypeTable typeTable, TypeRef type) {
+    return getNotImplementedString("SurfaceNamer.getLongRunningOperationTypeName");
+  }
+
+  /** The type name for the gPRC request. */
+  public String getRequestTypeName(ModelTypeTable typeTable, TypeRef type) {
+    return getNotImplementedString("SurfaceNamer.getRequestTypeName");
+  }
```

Comment:
I guess I could do the PHP implementation, but that assumes that the default uses the type nickname, rather than the full name. So far I have two uses of the method and they're too different to generalize from.